### PR TITLE
Make default JDBC read mappings reusable

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -464,8 +470,7 @@ protected void execute(Connection connection, String query)
         }
     }
 
-    @Override
-    public Optional<ReadMapping> toPrestoType(JdbcTypeHandle typeHandle)
+    public static Optional<ReadMapping> jdbcTypeToPrestoType(JdbcTypeHandle typeHandle)
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: Consider moving outside of `BaseJdbcClient`

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/StandardReadMappings.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public static ReadMapping decimalReadMapping(DecimalType decimalType)
         return sliceReadMapping(decimalType, (resultSet, columnIndex) -> encodeScaledValue(resultSet.getBigDecimal(columnIndex)));
     }
 
-    protected static ReadMapping charReadMapping(CharType charType)
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/StandardReadMappings.java]: sorry for that

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java (hunk 1 above)
Comment (electrum) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: I thought about that but wanted to minimize the diff. I'll move it to `StandardReadMappings` in a separate commit.

Comment (findepi) [reply presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java]: Appreciated

**Pull request closed without merging**